secure_fw/core/tfm_core_mem_check.c

```diff
@@ -35,6 +36,23 @@ static enum tfm_status_e has_access_to_region(const void *p, size_t s,
     range_access_allowed_by_mpu =
                           cmse_check_address_range((void *)p, s, flags) != NULL;
 
+    if (!range_access_allowed_by_mpu) {
+        /*
+         * Verification failure may be due to address range crossing
+         * one or multiple IDAU boundaries. In this case request a
+         * platform-specific check for access permissions.
+         */
+        cmse_address_info_t addr_info_base = cmse_TT((void *)p);
+        cmse_address_info_t addr_info_last = cmse_TT((void *)((uint32_t)p + s - 1));
+
+        if ((addr_info_base.flags.idau_region_valid != 0) &&
```

Comment:
I guess you'll add some `#if` here, to not force all platforms to provide an implementation of `tfm_spm_hal_has_access_to_region()`.

Reply:
Well, @anangl this is a good point. I don't know what the tf-m upstream project favors here. 
One option is what you suggest - an #ifdef IDAU_PRESENT, or sthg like this. 

The other option is to keep the check for _all_ platforms, regardless of IDAU existence. If the IDAU does not exist, then the idau_region_valid bits are zero, and we do not enter the if. For likning purposes, of course, we need to provide a __weak empty implementation of this returning false. This is only for linking: this function won't ever enter.
What do you think?

Also a question to @oyvindronningstad .

Reply:
> The other option is to keep the check for _all_ platforms, regardless of IDAU existence. If the IDAU does not exist, then the idau_region_valid bits are zero, and we do not enter the if.

Yes, this will be a dead code. But the compiler won't be able to detect it. To me it feels natural that there should be a possibility to get rid of this part, so I think some `#if` is the way to go.

Reply:
then @anangl would you like something such as IDAU_PRESENT? This complies with ARM CMSIS macros like MPU_PRESENT, FPU_PRESENT.

There is another option: use `#ifdef SAU_REGION_PRESENT == 0`.
SAU_REGION_PRESENT is defined in the SoC CMSIS headers. In nRF MDK this is defined as zero, indicating that there is no SAU and, implicitly, there is an implementation-defined IDAU. I guess if SAU exists, there is no IDAU ( cannot imagine a SoC that has both).

Reply:
Yes, that option with SAU regions present symbol sounds reasonable. However, it looks like it should be `__SAUREGION_PRESENT`, not `__SAU_REGION_PRESENT` like it is currently defined in MDK. :)

Reply:
Good point. We should use the right CMSIS header then. Anyway, for nRF this should be _not_ defined or defined as zero

Reply:
Updated @anangl , thanks.